### DISCUSSION TITLE
Documentation for Embed

### DISCUSSION THREAD

**Discussion by master3395** [rabbitmq/rabbitmq-website#1575] [Q&A] [answered]
Hello,

I were looking for documentation on embedded messages, but I do not see any for embeds such as discord, roblox or any other places.
Do anyone know if there is any, or if RabbitMQ can only send plain messages?

RabbitMQ sent this plain message:
![image](https://user-images.githubusercontent.com/4199669/208269013-6a246be0-c640-4d63-924e-89934eca646a.png)

https://discordjs.guide/popular-topics/embeds.html#using-the-embed-constructor

**Comment by lukebakken** [comment, marked as answer]
Messages in RabbitMQ are binary blobs. You should serialize these "embed" objects such that they can be sent is binary data.

**Comment by michaelklishin** [comment]
All protocols RabbitMQ supports assume that the messages sent are opaque to the broker (server). As long as applications put in enough metadata for other applications to know how to de-serialize or otherwise interpret the data, it's all good.
